app/src/main/java/comcmput301f16t01/github/carrier/DriverViewRequestActivity.java

```diff
@@ -1,16 +1,138 @@
+                    //statusImageView.setBackgroundResource(R.color.confirmedStatus);
+                    break;
+                case (Request.COMPLETE):
+                    statusImageView.setImageResource(R.drawable.complete);
+                    //statusImageView.setBackgroundResource(R.color.completeStatus);
+                    break;
+                case (Request.PAID):
+                    statusImageView.setImageResource(R.drawable.paid);
+                    //statusImageView.setBackgroundResource(R.color.paidStatus);
+                    break;
+                case (Request.CANCELLED):
+                    statusImageView.setImageResource(R.drawable.cancel);
+                    //statusImageView.setBackgroundResource(R.color.cancelledStatus);
+                    break;
+
+            }
+        }
+    }
+
+    private TextView fareAmountTextView;
+    private TextView startLocationTextView;
+    private TextView descriptionTextView;
+    private TextView riderTextView;
+    private TextView driverTextView;
+
+    /**
+     * Will set the various views to display the correct information for the
+     * request
+     */
+    private void setViewIds() {
+        fareAmountTextView.setText(new MakeRequestActivity().formatFare(request.getFare()));
+        startLocationTextView.setText(request.getStart().toString());
+        descriptionTextView.setText((request.getDescription()));
+        riderTextView.setText(request.getRider().getUsername());
+        if (request.getConfirmedDriver() != null) {
+            driverTextView.setText(request.getConfirmedDriver().getUsername());
+        }
+        else {
+            driverTextView.setText("");
+        }
+    }
+
+    /**
+     * Will get the ViewIds from the R.java file.
+     */
+    private void getViewIds() {
+        fareAmountTextView = (TextView) findViewById(R.id.textView_$fareAmount);
+        startLocationTextView = (TextView) findViewById(R.id.textView_start);
+        descriptionTextView = (TextView) findViewById(R.id.textView_description);
+        riderTextView = (TextView) findViewById(R.id.textView_rider);
+        driverTextView = (TextView) findViewById(R.id.textView_driver);
+    }
+
+    public void MakeOffer(View v) {
+        RequestController rc = new RequestController();
+        // Can not make an offer on a request that has a confirmed driver.
+        // Can not make an offer on a request that you hae already made an offer on.
+        // Can not make an offer on a cancelled request.
+        if (request.getConfirmedDriver() != null || request.getOfferedDrivers().contains(loggedInUser) ||
+                request.getStatus() == Request.CANCELLED) {
+            AlertDialog.Builder adb = new AlertDialog.Builder(DriverViewRequestActivity.this);
+            adb.setTitle("Error: ");
+            adb.setMessage("Unable to make an offer on the request.");
+            adb.setPositiveButton("Ok", new DialogInterface.OnClickListener() {
```

Comment:
I think a more clear error message depending on what the error was would be better so that the user can know why they are unable to make an offer on a specific request. Like, tell them that they have already made an offer on the request if they have already or tell them that they can't offer on completed requests.